ir/instr.cpp

```diff
@@ -1280,4 +1280,52 @@ unique_ptr<Instr> Store::dup(const string &suffix) const {
   return make_unique<Store>(ptr, val, align);
 }
 
+
+void Memset::print(std::ostream &os) const {
+  os << "memset " << ptr << " align " << align << ", " << val << ", " << bytes;
+}
+
+StateValue Memset::toSMT(State &s) const {
+  // TODO: check following lines are correct
```

Comment:
I modified memset, memcpy and is_dereferenceable using your conditions.
is_disjoint doesn't need modifying because when bytes is 0, is_disjoint is already no ub.